PalasoUIWindowsForms/ImageToolbox/ImageToolboxControl.cs

```diff
@@ -218,11 +214,9 @@ private void listView1_SelectedIndexChanged(object sender, EventArgs e)
 
 				if (_currentControl != null)
 				{
-					GetImageFromCurrentControl();
-
 					_panelForControls.Controls.Remove(_currentControl);
 					((IImageToolboxControl) _currentControl).ImageChanged -= new EventHandler(imageToolboxControl_ImageChanged);
-					_currentControl.Dispose();
```

Comment:
We explicitly set our reference to null below.  If this is the last one out the door then the finalizer calls dispose.  It is not clear that this *is* the last one to finish, so calling Dispose explicitly is unwise IMO.  Especially considering that there's no unmanaged resources to dispose of.  I see why the Dispose was added, but in theory if there isn't a reference loop (which should be fixed with weak refs) then the managed wrappers should actually dispose of the unmanaged resources that they are managing.  There doesn't seem to be any reason to not allow the finalizer to run.  Its not that expensive an op in this case.